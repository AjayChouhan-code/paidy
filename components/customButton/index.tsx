import React from "react";
import { Text, TouchableOpacity } from "react-native";

import { CustomButtonProps } from "./types";
import styles from "./styles";

export const CustomButton: React.FC<CustomButtonProps> = ({
  title,
  onPress,
  buttonStyle,
  textStyle,
  disabled = false,
}) => {
  return (
    <TouchableOpacity
      onPress={onPress}
      style={[styles.button, buttonStyle, disabled && styles.disabledButton]}
      activeOpacity={0.7}
      disabled={disabled}
    >
      <Text style={[styles.text, textStyle]}>{title}</Text>
    </TouchableOpacity>
  );
};
