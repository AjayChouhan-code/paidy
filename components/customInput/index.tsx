import React from "react";
import { View, Text, TextInput } from "react-native";

import { CustomInputProps } from "./types";
import styles from "./styles";

export const CustomInput: React.FC<CustomInputProps> = ({
  label,
  style,
  error,
  ...textInputProps
}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.label}>{label}</Text>
      <TextInput style={[styles.input, style]} {...textInputProps} />
      {error && <Text style={styles.errorText}>{error}</Text>}
    </View>
  );
};
