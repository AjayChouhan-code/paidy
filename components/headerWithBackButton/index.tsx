import React from "react";
import { View, Text, TouchableOpacity } from "react-native";

import { useNavigation } from "@react-navigation/native";
import { Ionicons } from "@expo/vector-icons";

import { styles } from "./styles";

export const HeaderWithBackButton = ({ title }: { title: string }) => {
  const navigation = useNavigation();

  return (
    <View style={styles.headerContainer}>
      <TouchableOpacity
        onPress={() => navigation.goBack()}
        style={styles.backButton}
      >
        <Ionicons name="arrow-back" size={24} color="black" />
      </TouchableOpacity>
      <Text style={styles.title}>{title}</Text>
    </View>
  );
};

export default HeaderWithBackButton;
