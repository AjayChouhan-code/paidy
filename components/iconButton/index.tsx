import React from "react";
import { TouchableOpacity, Text, View } from "react-native";

import { IconButtonProps } from "./types";
import styles from "./styles";

export const IconButton: React.FC<IconButtonProps> = ({
  label,
  onPress,
  buttonStyle,
  children,
}) => {
  return (
    <TouchableOpacity style={[styles.button, buttonStyle]} onPress={onPress}>
      <View style={styles.iconWrapper}>{children}</View>
      <Text style={[styles.label]}>{label}</Text>
    </TouchableOpacity>
  );
};
