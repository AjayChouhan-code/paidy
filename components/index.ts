export * from "./taskLists";
export * from "./customInput";
export * from "./customButton";
export * from "./headerWithBackButton";
export * from "./taskCard";
export * from "./iconButton";
