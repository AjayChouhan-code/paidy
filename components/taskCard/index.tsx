import React from "react";
import { Text, TouchableOpacity, View } from "react-native";

import { AntDesign } from "@expo/vector-icons";

import styles from "./styles";
import { TasKCard } from "./types";
import { Colors } from "@theme";

export const TaskCard: React.FC<TasKCard> = ({ item, onPress }) => {
  return (
    <View>
      <TouchableOpacity
        onPress={() => {
          onPress(item);
        }}
        style={styles.container}
        activeOpacity={0.7}
      >
        <View style={styles.containerTitle}>
          <Text numberOfLines={1} style={styles.title}>
            {item.title}
          </Text>
          <Text numberOfLines={2} style={styles.description}>
            {item.description}
          </Text>
        </View>
        <AntDesign name="right" size={24} color={Colors.dark_grey} />
      </TouchableOpacity>
    </View>
  );
};
