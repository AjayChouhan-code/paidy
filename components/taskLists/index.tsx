import React from "react";
import { FlatList, ListRenderItem, View } from "react-native";

import { TaskListsProps } from "./type";
import { TASK } from "@types";
import { TaskCard } from "../taskCard";

export const TaskLists: React.FC<TaskListsProps> = ({ taskList, onPress }) => {
  const renderItem: ListRenderItem<TASK> = ({ item }) => {
    return <TaskCard item={item} onPress={onPress} />;
  };

  return (
    <View>
      <FlatList
        data={taskList}
        keyExtractor={(item: TASK) => item?.id?.toString()}
        renderItem={renderItem}
      />
    </View>
  );
};
