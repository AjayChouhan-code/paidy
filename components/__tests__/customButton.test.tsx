import React from "react";
import { TouchableOpacity } from "react-native";
import renderer from "react-test-renderer";
import { CustomButton } from "@/components";

describe("CustomButton Component", () => {
  it("renders correctly with default props", () => {
    const tree = renderer
      .create(<CustomButton title="Click Me" onPress={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with custom styles", () => {
    const tree = renderer
      .create(
        <CustomButton
          title="Styled Button"
          onPress={() => {}}
          buttonStyle={{ backgroundColor: "blue" }}
          textStyle={{ color: "white" }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("disables the button when disabled prop is true", () => {
    const tree = renderer
      .create(
        <CustomButton
          title="Disabled Button"
          onPress={() => {}}
          disabled={true}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onPress when pressed", () => {
    const mockOnPress = jest.fn();
    const component = renderer.create(
      <CustomButton title="Press Me" onPress={mockOnPress} />
    );
    const button = component.root.findByType(TouchableOpacity);
    button.props.onPress();
    expect(mockOnPress).toHaveBeenCalled();
  });
});
