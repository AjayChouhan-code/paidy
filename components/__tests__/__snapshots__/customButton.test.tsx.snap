// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton Component disables the button when disabled prop is true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#A9A9A9",
      "borderRadius": 8,
      "opacity": 1,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 14,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Disabled Button
  </Text>
</View>
`;

exports[`CustomButton Component renders correctly with default props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1352A6",
      "borderRadius": 8,
      "opacity": 1,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 14,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Click Me
  </Text>
</View>
`;

exports[`CustomButton Component renders with custom styles 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 8,
      "opacity": 1,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 14,
          "fontWeight": "bold",
        },
        {
          "color": "white",
        },
      ]
    }
  >
    Styled Button
  </Text>
</View>
`;
